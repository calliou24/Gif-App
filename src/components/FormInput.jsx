import React, {useState} from 'react';
import PropTypes from 'prop-types'



export const FormInput = ( {add, reset} ) => {

    const [targetValue , setTargetValue] = useState('')
    //important that this use state have the argument of an Empty String of something 
    //we dont want to have a undefined props inside our component
    function sendValue(e) {
        
        e.preventDefault()
        add( e.target.firstChild.value)
        e.target.firstChild.value = ""
        setTargetValue("")
    }
    function setTargetValueReset(e) {
        
        e.preventDefault()
        reset()

    }
    return (
        <div className='form-cont'> 
        <p>Search</p>
        <form  onSubmit={ e => sendValue( e )}>
            <input onChange={ e => setTargetValue( e.target.value)}placeholder="..."/>
            <p className="text-reflect">{targetValue}</p>
            <button type='submit' className='btn-add-cat'>Add</button>
            <button onClick={e => setTargetValueReset(e)} type='submit' className='btn-add-cat'>Reset</button>
        </form>
        </div>
    )
} 

FormInput.propTypes = { 
    add : PropTypes.func.isRequired,
    reset : PropTypes.func.isRequired
}