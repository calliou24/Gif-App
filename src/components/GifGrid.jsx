import React,{ useState, useEffect } from 'react'
import { GifComponent } from './GifComponent/GifComponent'
import { useFetchGifs } from '../hooks/useFetchGifs'
export const GifGrid = ({category}) => {

  const {data, loading} = useFetchGifs( category ) 
    //use effect nos permite ejecutar algo una sola vez al renderizar el componente
    //de manera que este no esta cambiando y haciendo la funcion cada que se actualizar 
    //por ejemplo en un useState, entrando en un bucle
    //la funcion que se necestia aqui con el nombre getData esta almacenada en una carpeta
    //para poder modularizar nuestro archivo
  return (
    <div className='gif-grid'>
        <h2 className='gif-grid-title animate__animated animate__flipInX'>{category}</h2>
        <ul className='ul-gif-grid'>
            {loading ? "Loading..." : data.map( ({id,title,url}) => 
              <GifComponent 
                  key={id}
                  title={title}
                  url={url}
              />)}
        </ul>
    </div>
  )
}
