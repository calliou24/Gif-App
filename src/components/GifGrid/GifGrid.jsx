import React from "react";
import { GifComponent } from "../GifComponent/GifComponent";
import { useFetchGifs } from "../../hooks/useFetchGifs";

import styles from './gif-grid.module.css'

export const GifGrid = ({ category, deleteCategory }) => {
  const { data, loading } = useFetchGifs(category);
  //use effect nos permite ejecutar algo una sola vez al renderizar el componente
  //de manera que este no esta cambiando y haciendo la funcion cada que se actualizar
  //por ejemplo en un useState, entrando en un bucle
  //la funcion que se necestia aqui con el nombre getData esta almacenada en una carpeta
  //para poder modularizar nuestro archivo

  return (
    <div className={styles.gifCategory}>
      <div className={styles.titleCont}>  
      {loading
          ? "Loading..."
          : <h2 className={styles.title}>
        {category && category}
      </h2>}
      <button className={styles.deleteCat} onClick={() => deleteCategory(category)}> delete </button>
      </div>
      <ul className={styles.listGrid}>
        {data.map(({ id, title, url }) => (
              <GifComponent key={id} title={title} url={url} />
            ))}
      </ul>
    </div>
  );
};