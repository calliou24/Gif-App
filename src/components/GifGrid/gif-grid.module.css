.gifCategory {
  display: grid;
  row-gap: 40px;
}

.titleCont {
  display: flex;
  align-items: center;
  column-gap: 10px;
}

.title {
  font-size: 2rem;
}

.deleteCat{
  background:none;
  border: none;
  outline: none;
  font-size: 1rem;
  color: red;
  text-align: center;
  cursor: pointer;
  background:var(--secondary-color);
  padding:10px;
}

.deleteCat:hover {
  opacity: .8;
}
.listGrid{
  list-style: none;
  display: grid;
  place-items: center;
  grid-template-columns: repeat(3, 1fr);

}

.checkIcon{
  color:greenyellow;
}

@media (max-width : 50em){
  
.listGrid{
  list-style: none;
  display: grid;
  place-items: center;
  grid-template-columns: repeat(1, 1fr);

}
}