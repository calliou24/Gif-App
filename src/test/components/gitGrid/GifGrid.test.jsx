import { describe, expect, test } from "vitest";
import { shallow } from 'enzyme' 

import {GifGrid} from '../../../components/GifGrid'

describe("Should check for snapshots", ()=> {
    
    const cats = ["Anime"]
    
    const wrapper = shallow(<GifGrid category={cats}/>)

    test("Snapshot", ()=> {
        expect(wrapper).toMatchSnapshot()
    })      
})