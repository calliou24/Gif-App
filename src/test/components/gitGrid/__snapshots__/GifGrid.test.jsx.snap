// Vitest Snapshot v1

exports[`Should check for snapshots > Snapshot 1`] = `
<div
  className="gif-grid"
>
  <h2
    className="gif-grid-title animate__animated animate__flipInX"
  >
    Anime
  </h2>
  <ul
    className="ul-gif-grid"
  >
    Loading...
  </ul>
</div>
`;
